src/shared/reboot-util.c

```diff
@@ -110,14 +110,18 @@ int reboot_with_parameter(RebootFlags flags) {
 }
 
 int shall_restore_state(void) {
-        bool ret;
+        static int cached = -1;
+        bool b;
         int r;
 
-        r = proc_cmdline_get_bool("systemd.restore_state", /* flags = */ 0, &ret);
+        if (cached >= 0)
+                return cached;
+
+        r = proc_cmdline_get_bool("systemd.restore_state", PROC_CMDLINE_TRUE_WHEN_MISSING, &b);
         if (r < 0)
                 return r;
```

Comment:
When you touch the code, please make the failure non-critical, like fstab-util or battery-check do.